src/docfx/build/Build.cs

```diff
@@ -18,32 +18,44 @@ public static async Task Run(string docsetPath, CommandLineOptions options, Repo
             var repository = Repository.Create(docsetPath);
             Telemetry.SetRepository(repository?.Remote, repository?.Branch);
 
-            var locale = LocalizationUtility.GetLocale(repository, options);
-            var dependencyLock = LoadBuildDependencyLock(docsetPath, locale, repository, options);
+            var locale = LocalizationUtility.GetLocale(repository?.Remote, repository?.Branch, options);
+            var (dependencyLock, sourceRepoInfo) = LoadBuildDependencyLock(docsetPath, locale, repository, options);
 
             var restoreMap = RestoreMap.Create(dependencyLock);
+            var sourceRepo = sourceRepoInfo != default
+                ? Repository.Create(restoreMap.GetGitRestorePath(sourceRepoInfo.remote, sourceRepoInfo.branch, dependencyLock).path, sourceRepoInfo.branch, sourceRepoInfo.remote)
+                : default;
+
             try
             {
-                await Run(docsetPath, repository, locale, options, report, dependencyLock, restoreMap);
+                await Run(docsetPath, repository, locale, options, report, dependencyLock, restoreMap, sourceRepo);
             }
             finally
             {
                 restoreMap.Release();
             }
         }
 
-        private static async Task Run(string docsetPath, Repository repository, string locale, CommandLineOptions options, Report report, DependencyLockModel dependencyLock, RestoreMap restoreMap)
+        private static async Task Run(
+            string docsetPath,
+            Repository repository,
+            string locale,
+            CommandLineOptions options,
+            Report report,
+            DependencyLockModel dependencyLock,
+            RestoreMap restoreMap,
+            Repository sourceRepo = null)
```

Comment:
consider name it `fallbackRepo` or `locFallbackRepo`. `sourceRepo` could also mean source/target repo for a pull request.

Reply:
good idea